src/superannotate/lib/app/interface/base_interface.py

```diff
@@ -1,14 +1,140 @@
+
+    @property
+    @abstractmethod
+    def token(self):
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def logger(self):
+        raise NotImplementedError
+
+
+class Tracker:
+    TEAM_DATA = None
+    INITIAL_EVENT = {"event_name": "SDK init", "properties": {}}
+    INITIAL_LOGGED = False
+
+    @staticmethod
+    def get_mp_instance() -> Mixpanel:
+        # if "api.annotate.online" in get_default_controller()._backend_url:
+        #     return Mixpanel("ca95ed96f80e8ec3be791e2d3097cf51")
+        return Mixpanel("e741d4863e7e05b1a45833d01865ef0d")
+
+    @staticmethod
+    def get_default_payload(team_name, user_id, project_name=None):
+        return {
+            "SDK": True,
+            "Paid": True,
+            "Team": team_name,
+            "Team Owner": user_id,
+            "Project Name": project_name,
+            "Project Role": "Admin",
+            "Version": __version__,
+        }
+
+    def __init__(self, function):
+        self.function = function
+        functools.update_wrapper(self, function)
+
+    @staticmethod
+    def extract_arguments(function, *args, **kwargs) -> dict:
+        bound_arguments = signature(function).bind(*args, **kwargs)
+        bound_arguments.apply_defaults()
+        return dict(bound_arguments.arguments)
+
+    @staticmethod
+    def default_parser(function_name: str, kwargs: dict) -> tuple:
+        properties = {}
+        for key, value in kwargs.items():
+            if key == "self":
+                continue
+            elif key == "project":
+                properties["project_name"], properties["folder_name"] = extract_project_folder(value)
+            elif isinstance(value, str) and key == "project":
+                properties["project_name"] = value.split()
+            if isinstance(value, (str, int, float, bool, str)):
+                properties[key] = value
+            elif isinstance(value, dict):
+                properties[key] = value.keys()
+            elif isinstance(value, Sized):
+                properties[key] = len(value)
+            elif isinstance(value, Iterable):
+                properties[key] = "N/A"
+            else:
+                properties[key] = str(value)
+        return function_name, properties
+
+    def track(self, args, kwargs, success: bool, session):
+        try:
+            function_name = self.function.__name__ if self.function else ""
+            arguments = self.extract_arguments(self.function, *args, **kwargs)
+            event_name, properties = self.default_parser(function_name, arguments)
+
+            user_id = self.team_data.creator_id
+            team_name = self.team_data.name
+            properties["Success"] = success
+
+            default = self.get_default_payload(
+                team_name=team_name,
+                user_id=user_id,
+                project_name=properties.pop("project_name", None),
+            )
+            if "pytest" not in sys.modules:
```

Comment:
If we're tracking only when `"pytest" not in sys.modules` then can't we move the `if` to Line 128 and not do the rest of the processing? (Lines 90-102))

Reply:
This part also needs to be refactored
There are currently no tests for "mixpanel", but will be added.